### PR TITLE
feat(pasma-theme-builder): Add save and load draft themes

### PR DESCRIPTION
**Pull request by neretin-trike**
Добавил возможность автосохранения схем в `sessionStorage`. 
Теперь можно создавать новые схемы или продолжать работать со старыми. 
После создания пул-реквеста тема будет удалена.

<details>
  <summary>:baby_chick: Download canary assets:</summary>
  <blockquote>
    <details>
      <summary>color</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_ios-swift--canary.612.5598256449.swift'>color_asdk_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_kotlin--canary.612.5598256449.kt'>color_asdk_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_react-native--canary.612.5598256449.ts'>color_asdk_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_asdk_xml--canary.612.5598256449.xml'>color_asdk_xml--canary.612.5598256449.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_ios-swift--canary.612.5598256449.swift'>color_plasma_b2c_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_kotlin--canary.612.5598256449.kt'>color_plasma_b2c_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_react-native--canary.612.5598256449.ts'>color_plasma_b2c_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_plasma_b2c_xml--canary.612.5598256449.xml'>color_plasma_b2c_xml--canary.612.5598256449.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.612.5598256449.swift'>color_sberHealth_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.612.5598256449.kt'>color_sberHealth_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.612.5598256449.ts'>color_sberHealth_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_xml--canary.612.5598256449.xml'>color_sberHealth_xml--canary.612.5598256449.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.612.5598256449.swift'>color_sbermarket_business_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.612.5598256449.kt'>color_sbermarket_business_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.612.5598256449.ts'>color_sbermarket_business_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_xml--canary.612.5598256449.xml'>color_sbermarket_business_xml--canary.612.5598256449.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.612.5598256449.swift'>color_sbermarket_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.612.5598256449.kt'>color_sbermarket_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.612.5598256449.swift'>color_sbermarket_metro_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.612.5598256449.kt'>color_sbermarket_metro_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.612.5598256449.ts'>color_sbermarket_metro_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_xml--canary.612.5598256449.xml'>color_sbermarket_metro_xml--canary.612.5598256449.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.612.5598256449.ts'>color_sbermarket_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.612.5598256449.swift'>color_sbermarket_selgros_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.612.5598256449.kt'>color_sbermarket_selgros_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.612.5598256449.ts'>color_sbermarket_selgros_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_xml--canary.612.5598256449.xml'>color_sbermarket_selgros_xml--canary.612.5598256449.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.612.5598256449.swift'>color_sbermarket_wlbusiness_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.612.5598256449.kt'>color_sbermarket_wlbusiness_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.612.5598256449.ts'>color_sbermarket_wlbusiness_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_xml--canary.612.5598256449.xml'>color_sbermarket_wlbusiness_xml--canary.612.5598256449.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_xml--canary.612.5598256449.xml'>color_sbermarket_xml--canary.612.5598256449.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.612.5598256449.swift'>color_sberonline_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.612.5598256449.kt'>color_sberonline_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.612.5598256449.ts'>color_sberonline_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_xml--canary.612.5598256449.xml'>color_sberonline_xml--canary.612.5598256449.xml</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.612.5598256449.swift'>color_sberprime_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.612.5598256449.kt'>color_sberprime_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.612.5598256449.ts'>color_sberprime_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_xml--canary.612.5598256449.xml'>color_sberprime_xml--canary.612.5598256449.xml</a><br>
      </blockquote>
    </details>
    <details>
      <summary>shadow</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.612.5598256449.ts'>shadow_sbermarket_react-native--canary.612.5598256449.ts</a><br>
      </blockquote>
    </details>
    <details>
      <summary>typo</summary>
      <blockquote>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.612.5598256449.swift'>typo_mage_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.612.5598256449.kt'>typo_mage_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.612.5598256449.ts'>typo_mage_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.612.5598256449.swift'>typo_plasma_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.612.5598256449.kt'>typo_plasma_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.612.5598256449.ts'>typo_plasma_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.612.5598256449.swift'>typo_ruler_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.612.5598256449.kt'>typo_ruler_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.612.5598256449.ts'>typo_ruler_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.612.5598256449.swift'>typo_sage_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.612.5598256449.kt'>typo_sage_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.612.5598256449.ts'>typo_sage_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.612.5598256449.swift'>typo_sbermarket_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.612.5598256449.kt'>typo_sbermarket_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.612.5598256449.ts'>typo_sbermarket_react-native--canary.612.5598256449.ts</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.612.5598256449.swift'>typo_soulmate_ios-swift--canary.612.5598256449.swift</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.612.5598256449.kt'>typo_soulmate_kotlin--canary.612.5598256449.kt</a><br>
        <a href='https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.612.5598256449.ts'>typo_soulmate_react-native--canary.612.5598256449.ts</a><br>
      </blockquote>
    </details>
  </blockquote>
</details>

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.220.0-canary.612.5598256449.0
  npm install @salutejs/plasma-core@1.125.0-canary.612.5598256449.0
  npm install @salutejs/plasma-hope@1.220.0-canary.612.5598256449.0
  npm install @salutejs/plasma-icons@1.154.0-canary.612.5598256449.0
  npm install @salutejs/plasma-temple@1.169.0-canary.612.5598256449.0
  npm install @salutejs/plasma-tokens-b2b@1.24.0-canary.612.5598256449.0
  npm install @salutejs/plasma-tokens-b2c@0.33.0-canary.612.5598256449.0
  npm install @salutejs/plasma-tokens-web@1.39.0-canary.612.5598256449.0
  npm install @salutejs/plasma-tokens@1.58.0-canary.612.5598256449.0
  npm install @salutejs/plasma-ui@1.201.0-canary.612.5598256449.0
  npm install @salutejs/plasma-web@1.220.0-canary.612.5598256449.0
  npm install @salutejs/plasma-cy-utils@0.65.0-canary.612.5598256449.0
  npm install @salutejs/plasma-sb-utils@0.123.0-canary.612.5598256449.0
  npm install @salutejs/plasma-tokens-utils@0.29.0-canary.612.5598256449.0
  # or 
  yarn add @salutejs/plasma-b2c@1.220.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-core@1.125.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-hope@1.220.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-icons@1.154.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-temple@1.169.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-tokens-b2b@1.24.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-tokens-b2c@0.33.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-tokens-web@1.39.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-tokens@1.58.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-ui@1.201.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-web@1.220.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-cy-utils@0.65.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-sb-utils@0.123.0-canary.612.5598256449.0
  yarn add @salutejs/plasma-tokens-utils@0.29.0-canary.612.5598256449.0
  ```
</details>

### CODE REVIEW

**Review by Yakutoc — APPROVED** [review]

File: website/plasma-theme-builder/src/components/Generator/Generator.tsx [hunk 1]
```diff
@@ -116,8 +139,19 @@ export const Generator = ({ onPreviewTheme }: GeneratorProps) => {
...
 
+        if (savedThemes.some((savedTheme) => themeName === savedTheme.value)) {
+            setInputState({
+                status: 'error',
+                helperText: HELPER_TEXT_SAME_NAME_ERROR,
+            });
+            return;
+        }
+
         if (!RegExp(/^[a-z\d_]{3,}$/).test(themeName)) {
```
Comment (Yakutoc) [inline website/plasma-theme-builder/src/components/Generator/Generator.tsx:150]: @neretin-trike Оставь здесь пожалуйста комментарий, что делает это регулярка. )

**Review by kayman233 — APPROVED** [review]

File: website/plasma-theme-builder/src/components/Generator/Generator.tsx (hunk 1 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/Generator/Generator.tsx:150]: вместо комментария создал поясняющую переменную

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-612/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-612/

File: website/plasma-theme-builder/src/components/Generator/Generator.tsx [hunk 2]
```diff
@@ -31,7 +31,16 @@ const StyledTextField = styled(TextField)`
 const StyledButton = styled(Button)`
     display: block;
     margin: 2rem 0;
-    margin-left: auto;
+`;
+
+const StyledButtons = styled.div`
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/components/Generator/Generator.tsx:36]: хочется buttonGroup иметь =/
чтобы можно было ему задать отступы внутри причем стандартные =/

File: website/plasma-theme-builder/src/utils/themes.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    const savedThemeData = window.sessionStorage.getItem(`${THEME_BUILDER_PREFIX}-${themeName}`);
+    return savedThemeData ? JSON.parse(savedThemeData) : undefined;
+};
+
+export const loadAllThemeNames = (): (readonly [string, Date])[] => {
+    const themes = Object.keys(sessionStorage)
+        .filter((key) => key.startsWith(THEME_BUILDER_PREFIX))
+        .map((item) => {
+            const name = item.replace(`${THEME_BUILDER_PREFIX}-`, '');
+            const date = loadTheme(item)?.date || new Date();
```
Comment (Yeti-or) [inline website/plasma-theme-builder/src/utils/themes.tsx:34]: а если тут ошибка будет при добычи темы из хранилища?

File: website/plasma-theme-builder/src/components/Generator/Generator.tsx (hunk 2 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/Generator/Generator.tsx:36]: да, неплохо было бы такой компонент иметь, т.к. часто приходится такие штуки делать самому

File: website/plasma-theme-builder/src/components/Generator/Generator.tsx [hunk 4]
```diff
@@ -138,23 +173,69 @@ export const Generator = ({ onPreviewTheme }: GeneratorProps) => {
...
+                    <Form onSubmit={onSubmit}>
+                        <StyledSelectSavedThemes
+                            value={data.savedTheme}
+                            items={savedThemes}
+                            onChange={onChangeSelect('savedTheme')}
+                        />
+                        <StyledButtons>
+                            <StyledButton type="button" text="Удалить тему" onClick={onDeleteTheme} />
+                            <StyledButton
+                                type="button"
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/Generator/Generator.tsx:223]: вообще по-идее по умолчанию

File: website/plasma-theme-builder/src/utils/themes.tsx (hunk 3 above)
Comment (neretin-trike) [reply website/plasma-theme-builder/src/utils/themes.tsx:34]: ну это может случится только если пользователь ручками зачем-то в sessionStorage сломал название темы, и в таком случае тема просто будет пустая

File: website/plasma-theme-builder/src/components/Generator/Generator.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply website/plasma-theme-builder/src/components/Generator/Generator.tsx:223]: тогда почему приходится проставлять?

File: website/plasma-theme-builder/src/utils/themes.tsx (hunk 3 above)
Comment (Yeti-or) [reply website/plasma-theme-builder/src/utils/themes.tsx:34]: окей тогда

File: website/plasma-theme-builder/src/components/Generator/Generator.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (neretin-trike) [reply website/plasma-theme-builder/src/components/Generator/Generator.tsx:223]: да это я просто копипастнул кнопку из другого места на этой же странице. А на тот момент, когда я её добавлял, у нас по умолчанию type не стоял. В принципе могу убрать

**Review by Yeti-or — APPROVED** [review]

**Comment by Salute-Eva** [comment]
🚀 This PR is included in version: @salutejs/plasma-b2c@1.220.0, @salutejs/plasma-core@1.125.0, @salutejs/plasma-cy-utils@0.65.0, @salutejs/plasma-hope@1.220.0, @salutejs/plasma-icons@1.154.0, @salutejs/plasma-sb-utils@0.123.0, @salutejs/plasma-temple-docs@0.160.0, @salutejs/plasma-temple@1.169.0, @salutejs/plasma-tokens-b2b@1.24.0, @salutejs/plasma-tokens-b2c@0.33.0, @salutejs/plasma-tokens-native@1.9.0, @salutejs/plasma-tokens-utils@0.29.0, @salutejs/plasma-tokens-web@1.39.0, @salutejs/plasma-tokens@1.58.0, @salutejs/plasma-typo@0.36.0, @salutejs/plasma-ui-docs@0.209.0, @salutejs/plasma-ui@1.201.0, @salutejs/plasma-web-docs@0.178.0, @salutejs/plasma-web@1.220.0, @salutejs/plasma-website@0.186.0 🚀

**Pull request merged**